spinnaker/spinnaker_system/google_http_lb_upsert_test.py

```diff
@@ -0,0 +1,894 @@
+
+    return st.OperationContract(
+      self.new_post_operation(title='upsert min http lb', data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+
+  def delete_load_balancer(self):
+    '''Deletes the L7 LB.
+    '''
+    bindings = self.bindings
+    delete = {
+      'type': 'deleteLoadBalancer',
+      'cloudProvider': 'gce',
+      'loadBalancerType': 'HTTP',
+      'loadBalancerName': self.__lb_name,
+      'region': bindings['TEST_GCE_REGION'],
+      'regions': [bindings['TEST_GCE_REGION']],
+      'credentials': bindings['GCE_CREDENTIALS'],
+      'user': '[anonymous]'
+    }
+
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[delete],
+      description='Delete L7 Load Balancer: {0} in {1}:{2}'.format(
+        self.__lb_name,
+        bindings['GCE_CREDENTIALS'],
+        bindings['TEST_GCE_REGION'],
+      ),
+      application=self.TEST_APP
+    )
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    (contract_builder.new_clause_builder('Health Check Removed', retryable_for_secs=30)
+     .list_resource('httpHealthChecks')
+     .excludes_path_value('name', self.__proto_hc['name'])
+    )
+    (contract_builder.new_clause_builder('Url Map Removed',
+                                         retryable_for_secs=30)
+     .list_resource('urlMaps')
+     .excludes_path_value('name', self.__lb_name)
+    )
+    (contract_builder.new_clause_builder('Forwarding Rule Removed',
+                                         retryable_for_secs=30)
+     .list_resource('globalForwardingRules')
+     .excludes_path_value('name', self.__lb_name)
+    )
+
+    return st.OperationContract(
+      self.new_post_operation(
+        title='delete_load_balancer', data=payload, path='tasks'),
+      contract=contract_builder.build())
+
+
+  def change_health_check(self):
+    '''Changes the health check associated with the LB.
+    '''
+    upsert = copy.deepcopy(self.__proto_upsert)
+    hc = copy.deepcopy(self.__proto_hc)
+    hc['requestPath'] = '/hello'
+    hc['port'] = 8080
+    self._set_all_hcs(upsert, hc)
+
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[upsert],
+      descriptiopn='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    hc_link = self._make_hc_link(hc['name'])
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    self._make_contract_clauses(contract_builder, upsert)
+
+    return st.OperationContract(
+      self.new_post_operation(title='change health checks',
+                              data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+
+  def change_backend_service(self):
+    '''Changes the default backend service associated with the LB.
+    '''
+    hc = copy.deepcopy(self.__proto_hc)
+    bs_upsert = copy.deepcopy(self.__proto_upsert)
+    hc['name'] = 'updated-hc'
+    hc['requestPath'] = '/hello'
```

Comment:
Could this changed health check and the previous one be confused somehow (a bug where you updated the wrong thing here and updated what the previous test was checking for, or vice versa)? If so then you might want to make these different so that the failures arent masked by the other test.